base-notebook/start.sh

```diff
@@ -2,6 +2,34 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
+## TEMP DOCS:
+#
+# - Will a PATH env from the docker's root environment, that is propegated to
+#   the user environment we switch into, be a security escalation? It seems to
+#   have been the intention so far looking in script.sh, even though it turned
+#   out that secure_path had the final say which overrode other parts.
+# - Will linux user ids always be either not coupled, or uniquely coupled, to a
+#   username?
+# - Is it correct that the filesystem will reference permission to user IDs, and
+#   the /etc/passwd file will couple these IDs to usernames?
+# - Can multiple usernames be coupled to the userid 0?
+# - In this code, we see a big if/else clause about "id --uid == 0", but "id
+#   --gid == 0" would be quite powerful as well right? Would it make sense to
+#   check for either "id --uid == 0" or "id --gid == 0"?
```

Comment:
@rkdarst and @maresb, perhaps you can help me understand some of these questions I've raised above?

Reply:
```
# - Can multiple usernames be coupled to the userid 0?
# - In this code, we see a big if/else clause about "id --uid == 0", but "id
#   --gid == 0" would be quite powerful as well right? Would it make sense to
#   check for either "id --uid == 0" or "id --gid == 0"? 
```

I think the user management (`usermod`, `mv /home/jovyan ...`) won't work with just gid=0 ... it needs to be uid=0 to have those permissions.  Backed up by a quick but not very complete test...